src/main/java/com/snowflake/kafka/connector/internal/SnowflakeTelemetryPipeStatus.java

```diff
@@ -0,0 +1,181 @@
+  static final String COMMITTED_OFFSET  = "committed_offset";
+  static final String PURGED_OFFSET     = "purged_offset";
+
+  // Legacy metrix
+  AtomicLong totalNumberOfRecord;       // total number of record
+  AtomicLong totalSizeOfData;           // total size of data
+  static final String RECORD_NUMBER     = "record_number";
+  static final String BYTE_NUMBER       = "byte_number";
+
+  // File count info
+  AtomicLong fileCountOnStage;                  // files that are currently on stage
+  AtomicLong fileCountOnIngestion;              // files that are being ingested
+  AtomicLong fileCountPurged;                   // files that are purged
+  AtomicLong fileCountTableStage;               // files that are moved to table stage
+  static final String FILE_COUNT_ON_STAGE       = "file_count_on_stage";
+  static final String FILE_COUNT_ON_INGESTION   = "file_count_on_ingestion";
+  static final String FILE_COUNT_PURGED         = "file_count_purged";
+  static final String FILE_COUNT_TABLE_STAGE    = "file_count_table_stage";
+
+  // Cleaner restart count
+  AtomicLong cleanerRestartCount;               // how many times the cleaner restarted
+  static final String CLEANER_RESTART_COUNT     = "cleaner_restart_count";
+
+  //------------ following metrics are not cumulative, reset every time sent ------------//
+  // Average lag of Kafka
+  AtomicLong averageKafkaLag;                           // average lag on Kafka side
+  AtomicLong averageKafkaLagRecordCount;                // record count
+  static final String AVERAGE_KAFKA_LAG                 = "average_kafka_lag";
+  static final String AVERAGE_KAFKA_LAG_RECORD_COUNT    = "average_kafka_lag_record_count";
+
+  // Average lag of ingestion
+  AtomicLong averageIngestionLag;                       // average lag between file upload and file delete
+  AtomicLong averageIngestionLagFileCount;              // file count
+  static final String AVERAGE_INGESTION_LAG             = "average_ingestion_lag";
+  static final String AVERAGE_INGESTION_LAG_FILE_COUNT  = "average_ingestion_lag_file_count";
+
+  // Average lag of commit
+  AtomicLong averageCommitLag;                          // average lag between file upload and ingest api calling
+  AtomicLong averageCommitLagFileCount;                 // file count
+  static final String AVERAGE_COMMIT_LAG                = "average_commit_lag";
+  static final String AVERAGE_COMMIT_LAG_FILE_COUNT     = "average_commit_lag_file_count";
+
+  // Need to update two values atomically when calculating lag, thus
+  // a lock is required to protect the access
+  private final Lock lagLock;
+
+  AtomicLong startTime;             // start time of the status recording period
+  static final String START_TIME    = "start_time";
+  static final String END_TIME      = "end_time";
+
+  SnowflakeTelemetryPipeStatus(final String tableName, final String stageName, final String pipeName)
+  {
+    this.tableName = tableName;
```

Comment:
seems like some variables can be constant. could you change them to `final`?

Reply:
ack, will change that